### PR TITLE
feat(styles): add skeleton sass modules and stories

### LINKED ISSUE #9130: Skeleton sass module refactor
This issue is for updating the **Skeleton** component to Sass Modules in the `@carbon/styles` package. In this refactor, we will:

- [ ] Copy over styles for for the component to `packages/styles/scss/components/search/_{component}.scss`
  - Any additional files will live in the folder `packages/styles/scss/components/{component}/`
- [ ] Create an `_index.scss` at `packages/styles/scss/components/{component}/_index.scss`
- [ ] Create a test file at `packages/styles/scss/components/__tests__/{component}-test.js`
- [ ] add the storybook stories to @carbon/react

The `_index.scss` should:

- `@forward` the mixin and any variables from `_{component}.scss`
- Include the styles from `_{component}.scss` by calling the mixin

The `search-test.js` file should test the following:

- The Public API of the `{component}/_index.scss` file
- Any configuration options that are present for the file

The `_{component}.scss` will then need to be refactored to use the new import paths and conventions. You can find common recipes here: https://github.com/carbon-design-system/carbon/issues/8899#issuecomment-860887959

### Migration process

- Find the component SCSS file from `packages/components`
- Create an `_index.scss` file for this component over in `packages/styles/scss/components/<component>/_index.scss`
- Copy over on the `*.scss` files, ignore the JS/HBS/md files for the most part, into this folder
- Update the `_<component>.scss` file with the following changes:
  - Remove the `import-once` import
  - Remove the `exports` mixin that is used
  - Update `_index.scss` to `@use '<component>';` and include the mixin afterwards

```scss
@use 'example-component';

@include example-component.example-component;
```

- Update the imports of the file from `@import` to `@use`
  - `css--reset` -> `utilities/component-reset`
  - Duration tokens coming from `_motion.scss` and have a name change
  - `rem` and `em` helpers come from `../utilities/convert`
  - `carbon--spacing` -> `spacing`
  - `carbon--breakpoint` -> `breakpoint`
  - Update tokens to use CSS Custom Properties
- Update `scss/components/_index.scss` entrypoint
- Update any calculations on tokens to use `calc()`

### Common migrations

**`$prefix`**

```diff
+@use '../../config' as *;
```

**`type-style`**

```diff
+@use '../../type';

...

.#{$prefix}--component {
-  @include type-style('body-short-01');
+  @include type.type-style('body-short-01');
}
```

**Button reset**

```diff
-@import '../../globals/scss/helper-mixins';
+@use '../../utilities/button-reset';
```

**Component reset**

```diff
-@import '../../globals/scss/css--reset';
+@use '../../utilities/component-reset';
```

**Tokens**

```diff
-@import '../../globals/scss/vars';
+@use '../../theme' as *;
```

**`duration--*`**

```diff
-@import '../../globals/scss/vars';
@use '../../motion' as *;

...

.#{$prefix}--my-component {
-  transition: color $duration--fast-01 motion(standard, productive);
+  transition: color $duration-fast-01 motion(standard, productive);
}
```

**helper mixins**

```diff
-@import '../../globals/scss/helper-mixins';

// Focus outline
+@use '../../utilities/focus-outline' as *;
```

**spacing**

```diff
-@import '../../globals/scss/vars';
+@use '../../spacing' as *;
```

**rem**

```diff
+@use '../../utilities/convert' as *;
```

### PR DESCRIPTION
**Pull request by jnm2377**
Closes #9130 

**New**

- adds skeleton styles to sass modules
- adds `SkeletonPlaceholder` and `SkeletonText` to `@carbon/react` stories

#### Testing / Reviewing
- ensure all test pass
- look at the carbon-next deploy preview and go to Skeleton stories, verify that styles are rendering correctly (can cross reference current storybook styles)

### CODE REVIEW

File: packages/styles/scss/components/_index.scss [hunk 1]
```diff
@@ -41,6 +41,7 @@
 @use 'radio-button';
 @use 'search';
 @use 'select';
+@use 'skeleton' as skeleton-styles;
```
Comment (jnm2377) [inline packages/styles/scss/components/_index.scss]: note for reviewers: without `as skeleton-styles` the it was giving me an error about there already being a skeleton file (`data-table/skeleton` line 19). Wasn't sure if this was ok to do.. or should I rename the file to be something else?

**Comment by sstrubberg** [comment]
image

Getting a fail on the skeleton test.

**Comment by jnm2377** [comment]
@tay1orjones looks like the percy snapshot for CodeSnippet is failing bc in the snapshot it's missing the show more button. I checked out the story and it's there, it just took a second to load. I'm not touching any codesnippet styles here. Any thoughts on what I should do?

**Comment by tay1orjones** [comment]
@jnm2377 interesting! My first thought is to modify the codesnippet vrt and make sure `cy.findByText('show more').should('be.visible')` before the cypress snapshot is taken

**Comment by jnm2377** [comment]
@tw15egan @sstrubberg sorry for the delay, this is ready for re-review now.

**Review by sstrubberg — APPROVED** [review]

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

**Pull request merged**